tinygrad/codegen/linearizer.py

```diff
@@ -235,7 +236,7 @@ def global_load(self, i:int, idxs:Sequence[VariableOrNum], const=None) -> List[T
     if len(upcast_dim) == 1 and len(expanded_nodes[upcast_dim[0]]) in [4,2]:
       dim, amt = upcast_dim[0], len(expanded_nodes[upcast_dim[0]])
 
-    cache: Dict[str, Token] = {}
+    cache: Dict[str, Token] = load_cache[self.get_buffer_name(i)] if load_cache is not None and load_type == "val" else {}
```

Comment:
I'd do self.load_cache instead of passing it in. Like self.uop. And I don't think it should ever be None.

Reply:
Why would we want to cache accumulator definitions?